ttkwidgets/shell.py

```diff
@@ -0,0 +1,205 @@
+            self.text_update()
+            return
+
+    def on_configure(self, event):
+        padding = 4
+        width = self.master.winfo_width() - padding
+        height = self.master.winfo_height() - padding
+        self.config(width=width, height=height)
+        for t in self.texts:
+            self.itemconfig(t, width=width)
+        self.itemconfig(self.last_text, width=width)
+
+    def text_update(self):
+        """
+        Updates the text on the screen.
+        """
+        self.textvariable.set(self.buffer)
+        if self.last_text:
+            self.delete(self.last_text)
+        kwargs = {
+            'anchor': tk.NW,
+            'fill': 'white',
+            'text': self.buffer,
+            'width': self['width'],
+            'font': self.font,
+        }
+        self.last_text = self.create_text(*self.line_pos, **kwargs)
+        return
+
+    def cursor_blink(self):
+        if self.cursor:
+            self.delete(self.cursor)
+            self.cursor = None
+        if self.last_text is not None and self.blink:
+            pos = self.cursor_pos
+            font = self.itemcget(self.last_text, 'font').split()
+            width, height = self.max_char_width, int(font[-1])
+            pos = pos + (pos[0] + width, pos[1] + height)
+            self.cursor = self.create_rectangle(*pos, fill='white')
+        self.blink = not self.blink
+        self.after(1000, self.cursor_blink)
+
+    @property
+    def max_char_width(self):
+        """
+        Gets the width of a W character
+
+        :returns: width of W with the font the shell uses
+        :rtype: int
+        """
+        font = tkfont.Font(self, font=self.itemcget(self.last_text, 'font'))
+        return font.measure('W')
+
+    @property
+    def cursor_pos(self):
+        text = self.itemcget(self.last_text, 'text')
+        font = tkfont.Font(self, font=self.itemcget(self.last_text, 'font'))
+        text_len = font.measure(text)
+        width = self.max_char_width
+        span = self.text_line_span()
+        x = text_len % int(self['width']) + width + self.line_pos[0]
+        y = self.line_pos[1] + 15 * (span - 1)
+        return x, y
+
+    def text_line_span(self):
+        """
+        Gets the number of lines to display the current text
+
+        :returns: number of lines
+        :rtype: int
+        """
+        text = self.itemcget(self.last_text, 'text')
+        font = tkfont.Font(self, font=self.itemcget(self.last_text, 'font'))
+        text_len = font.measure(text)
+        n_lines = text_len // int(self['width']) + 1
+        return n_lines
+
+    def call_command(self, command, *args):
+        """
+        Calls command callbacks
+
+        :param command: Command name to call
+        :type command: str
+        """
+        commands = self.commands.get(command, [])
+        if callable(commands):
+            return commands(*args)
+
+        for callback in commands:
+            callback(*args)
+
```

Comment:
Perhaps adding something of a catch-all function would be useful? Or, if that does not exist, showing an error for unknown commands?